src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
+// const deer = new Herbivore('Bembi');
+// const panther = new Carnivore('Bagira');
+// const lion = new Carnivore('King');
+// const rabbit = new Herbivore('Max');
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  name = this.name;
+  health = this.health;
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden !== true) {
+      animal.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((element) => element.health > 0);
```

Comment:
[CODE PERFORMANCE] - The `filter` method is used correctly to remove dead animals from the `Animal.alive` array. However, it's not optimal to filter the array every time the `bite` method is called, as this can be performance-intensive with a large number of animals. Consider filtering the array only when an animal's health becomes less than or equal to 0.